mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/resources/ResourcesListFragment.kt

```diff
@@ -0,0 +1,327 @@
+import rs.readahead.washington.mobile.views.activity.viewer.PDFReaderActivity.Companion.VIEW_PDF
+import rs.readahead.washington.mobile.views.base_ui.BaseBindingFragment
+
+
+@AndroidEntryPoint
+class ResourcesListFragment :
+    BaseBindingFragment<FragmentResourcesListBinding>(FragmentResourcesListBinding::inflate) {
+
+    private val model: ResourcesViewModel by viewModels()
+
+    private var availableResources = HashMap<String, Resource>()
+    private var downloadedResources = HashMap<String, Resource>()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        if (!hasInitializedRootView) {
+            hasInitializedRootView = true
+            initView()
+        }
+        initObservers()
+        refreshLists()
+    }
+
+    override fun onResume() {
+        super.onResume()
+    }
+
+    private fun initObservers() {
+        with(model) {
+            resources.observe(
+                viewLifecycleOwner
+            ) { listFormResult ->
+                onAvailableResourcesList(
+                    listFormResult!!
+                )
+            }
+
+            savedResources.observe(viewLifecycleOwner) { resources ->
+                resources.forEach { downloadedResources[it.fileName] = it }
+                updateResourcesViews()
+            }
+
+            downloadedResource.observe(viewLifecycleOwner) {
+                downloadedResource(it)
+            }
+
+            pdfFile.observe(viewLifecycleOwner) {
+                openPdf(it)
+            }
+
+            progress.observe(viewLifecycleOwner) {
+                showProgress(it)
+            }
+
+            deletedResource.observe(viewLifecycleOwner) {
+                DialogUtils.showBottomMessage(
+                    baseActivity,
+                    getString(
+                        R.string.Resources_RemovedResource, it
+                    ),
+                    false
+                )
+            }
+
+            error.observe(viewLifecycleOwner) {
+                it?.message?.let { it1 ->
+                    DialogUtils.showBottomMessage(
+                        baseActivity,
+                        it1,
+                        true
+                    )
+                }
+            }
+        }
+    }
+
+    private fun showProgress(show: Boolean) {
+        if (show) {
+            binding.progressBar.show()
+        } else {
+            binding.progressBar.hide()
+        }
+    }
+
+    private fun updateResourcesViews() {
+        binding.blankResources.removeAllViews()
+        createResourcesViews(availableResources.values.toList(), binding.blankResources, false)
+        if (availableResources.isEmpty()) {
+            binding.availableResourcesEmpty.show()
+            binding.availableResourcesInfo.hide()
+        } else {
+            binding.availableResourcesInfo.show()
+            binding.availableResourcesEmpty.hide()
```

Comment:
Maybe, we could add a method "updateResourceList" containing the repetitive logic for showing or hiding views.